#Scenes/SceneScripts/credits.gd

```diff
@@ -0,0 +1,327 @@
+	"Pterion": [[Roles.Art, Roles.Design], sapling_type.Maid],
+	"ROBBERGON": [[Roles.Team_lead_design], sapling_type.Snow],
+	"Saphu": [[Roles.Art, Roles.Design], sapling_type.Sleepy],
+	"Sappysque": [[Roles.Art], sapling_type.Emo],
+	"Tomzkk": [[Roles.Programming, Roles.Design], sapling_type.Old],
+	"Tradgore": [[Roles.Writing, Roles.Design], sapling_type.Sleepy],
+	"Turtyo": [[Roles.Project_lead, Roles.Team_lead_programming, Roles.Design], sapling_type.Gamer],
+	"TyTy": [[Roles.Programming], sapling_type.Old],
+	"Vsiiesk ": [[Roles.Art], sapling_type.Nerd],
+	"Vyto (Vytonium)": [[Roles.Music], sapling_type.Old],
+	"Walles": [[Roles.Design], sapling_type.Sleepy],
+	"Zannmaster": [[Roles.Design], sapling_type.Milf],
+}
+
+## The time needed for an animation to reach the center of the screen [br]
+## This is especially useful for animations that are very fast and we need to delay them [br]
+## This lets the previous animation time to get out of the screen center
+const ANIMATION_TIME_TO_SCREEN_CENTER: Dictionary = {
+	"In/cool_in_1": 2.3,
+	"In/emo_in_1": 0, #enters by the left, will need to wait for previous out animation to completely finish
+	"In/gamer_in_1": 1.5,
+	"In/maid_in_1": 0.7,
+	"In/milf_in_1": 2.6,
+	"In/nerd_in_1": 1.7,
+	"In/old_in_1": 1.5,
+	"In/sleepy_in_1": 0.4,
+	"In/snow_in_1": 0.4,
+}
+
+## Time needed for an animation to free the space at the center of the screen [br]
+## Delay the previous animation by this duration if the time it takes to reach the center is smaller
+## than the time it takes the "out" animation to free the screen center
+const ANIMATION_TIME_OUT_SCREEN_CENTER: Dictionary = {
+	"Out/cool_out_1": 1,
+	"Out/emo_out_1": 2.1,
+	"Out/gamer_out_1": 1.3,
+	"Out/maid_out_1": 0.5,
+	"Out/milf_out_1": 0.6,
+	"Out/nerd_out_1": 0.7,
+	"Out/old_out_1": 0.9,
+	"Out/sleepy_out_1": 2.2,
+	"Out/snow_out_1": 1.1,
+}
+
+## Launch the loop of the scene
+func _ready() -> void:
+	# start by setting up various things
+	black_overlay.modulate.a = 0
+	name_role_label.text = ""
+	name_role_label.add_theme_color_override("font_color", COLOR_FADE_OUT)
+	_animation_loop()
+	
+	await animation_ended
+	
+	# fade to black and go to next scene
+	var tween: Tween = get_tree().create_tween()
+	tween.tween_property(black_overlay, "modulate:a", 1, 3)
+	
+	await tween.finished
+	
+	SceneManager.goto_scene("res://#Scenes/credits_extra.tscn")
+	
+	
+
+## The main animation loop, controls the entry timing of all the saplings [br]
+## This is done by controling the two animation player of the scene
+func _animation_loop() -> void:
+	# a little timer to start the scene
+	var timer: SceneTreeTimer = get_tree().create_timer(1)
+	var team_members_names: Array = _choose_member_order()
+
+	await timer.timeout
+	
+	# play first animation with animation player in
+	# this is needed since the loop also plays the out animation of the previous avatar
+	# but there is no previous avatar before the first one
+	var first_member: String = team_members_names[0]
```

Comment:
Should probably change to `first_member_namer` or something to make it more clear what this variable is

Reply:
indeed

Reply:
done in [f3edd60](https://github.com/Saplings-Projects/1M_sub/pull/116/commits/f3edd60c74d4a008653584ea998376a07bd3c904)